org.eclipse.m2e.pde/src/org/eclipse/m2e/pde/MavenTargetLocation.java

```diff
@@ -273,8 +285,67 @@ private boolean isPomType(Artifact artifact) {
 		return POM_PACKAGE_TYPE.equals(artifact.getExtension());
 	}
 
+	private File getFeatureFile(Artifact artifact, CacheManager cacheManager) {
+		File baseFile = artifact.getFile();
+		
+		if(baseFile == null) {
+			return null;
+		} else if (baseFile.isDirectory()) {
+			File featureFile = new File(baseFile, ICoreConstants.FEATURE_FILENAME_DESCRIPTOR);
+			return featureFile.exists() ? featureFile : null;
+		} else if (DEFAULT_PACKAGE_TYPE.equals(FilenameUtils.getExtension(baseFile.getName()))) {
+			File featureFile = unpackFeatureFile(artifact, cacheManager);
+			return featureFile;
```

Comment:
The variable can be in-lined.
```suggestion
			return unpackFeatureFile(artifact, cacheManager);
```